data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -181,11 +197,16 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
                     continue;
                 }
 
-                if (!excludeKeysSet.isEmpty() && excludeKeysSet.contains(key)) {
+                if (excludeKeysSet.contains(key)) {
                     LOG.debug(String.format("Key is being excluded: '%s'", key));
                     continue;
                 }
 
+                if (defaultKeysSet.contains(key)) {
```

Comment:
I'm not sure about this behavior. This seems to have the default key value overriding the actual value from the Event. This doesn't seem expected or desirable.